octodns/provider/plan.py

```diff
@@ -60,6 +60,12 @@ def __init__(
         self.update_pcent_threshold = update_pcent_threshold
         self.delete_pcent_threshold = delete_pcent_threshold
 
+        # Zone thresholds take precedence over provider
+        if existing and existing.update_pcent_threshold is not None:
+            self.update_pcent_threshold = existing.update_pcent_threshold
+        if existing and existing.delete_pcent_threshold is not None:
+            self.delete_pcent_threshold = existing.delete_pcent_threshold
+
```

Comment:
I think combining the two ways the pcents get set into one section makes the logic a slightly clearer, comment explained it well, this is just belt & suspenders:

```suggestion

        # Zone thresholds take precedence over provider
        if existing and existing.update_pcent_threshold is not None:
            self.update_pcent_threshold = existing.update_pcent_threshold
        else:
            self.update_pcent_threshold = update_pcent_threshold
        if existing and existing.delete_pcent_threshold is not None:
            self.delete_pcent_threshold = existing.delete_pcent_threshold
        else:
            self.delete_pcent_threshold = delete_pcent_threshold
```

Reply:
Good suggestion, that reads better.